src/musicbrainz/web/musicbrainzrecordingstask.cpp

```diff
@@ -202,6 +203,7 @@ void MusicBrainzRecordingsTask::doNetworkReplyFinished(
 }
 
 void MusicBrainzRecordingsTask::triggerSlotStart() {
+    slotIdle();
```

Comment:
I would add here an extra function for the purpose to receive timeout events. 
Than you can add some more checks here:
If it is already in the finish state, return, because it is a stray signal.
You expect the pending state right? 

You probably need a second idle state  that bypasses the assertion deeper in the code.

Reply:
When it is timed out, nothing bad happens because the task was finished and right away another task started. I am not sure, should we also check in here? 

I wanted to try actually, if the task is timed out it wasn't making the task idle. Before the timed out task started it was externally changed to Pending. So this was returning `Task is still busy and cannot be started again`.

Actually we need pending state just for aborting the task.
And we don't need pending state anymore because, if the state is pending the new task is not starting.

Reply:
But how does abbort() abbort the task when the timer running?

Reply:
Timer counts only if the task is too fast. When the timer counts, the state is terminated, so it can not abort. So making the state pending the task can aborted.

You are asking how does it abort when the Timer running right? I don't know actually. Since we abort the task, would it be abort the QTimer aswell so it won't emit to signal so next slot won't be called?